### PR TITLE
chore: remove organization requirement from convertGroup() 

### PR DESCRIPTION
**Pull request by Emyrk**
Removed role and org information from the Group's api. This information was always excessive (I even had a TODO comment to remove it). It was kept in because dealing with nested structs on the cli was difficult.

# Cli table formatter fixed

The cli table formatter used to not work well with nested structs. I fixed the `default_sort` bug which allows the User struct to be defined in the way this PR does.

I added the golden file for users list to track this change.

### CODE REVIEW

File: cli/testdata/coder_users_list.golden [hunk 1]
```diff
@@ -0,0 +1,3 @@
+USERNAME   EMAIL                CREATED AT            STATUS   
+testuser   testuser@coder.com   [timestamp]  active
```
Comment (mafredri) [inline cli/testdata/coder_users_list.golden:2]: Can we replace it with the same length string? Worried someone might think this is a bug. 😄

**Review by mafredri — APPROVED** [review]
Looks alright, but how is default sort order decided now that it's removed from the struct?

**Comment by Emyrk** [comment]
> Looks alright, but how is default sort order decided now that it's removed from the struct?

Default order is not removed, it's fixed. Prior to this PR you had to have `default_sort` set on nested structures because our code had an issue. With the fix, `default_sort` is set on the `username` in `MinimalUser`. So it's sorted by username

File: cli/testdata/coder_users_list.golden (hunk 1 above)
Comment (Emyrk) [reply cli/testdata/coder_users_list.golden:2]: It then looks like this:
```
USERNAME   EMAIL                CREATED AT            STATUS   
testuser   testuser@coder.com   [timestamp         ]  active   
testuser2  testuser2@coder.com  [timestamp         ]  dormant  

```

Comment (Emyrk) [reply cli/testdata/coder_users_list.golden:2]: And in json
```json
    "created_at": "[timestamp                ]",
    "last_seen_at": "[timestamp                ]",
```

**Comment by Emyrk** [comment]
timestamp padding moved here: https://github.com/coder/coder/pull/12256

**Pull request merged**